clisops/core/subset.py

```diff
@@ -534,7 +534,7 @@ def create_mask(
     if wrap_lons:
         warnings.warn("Wrapping longitudes at 180 degrees.")
 
-    if len(x_dim.shape) == 1 & len(y_dim.shape) == 1:
+    if len(x_dim.shape) == 1 and len(y_dim.shape) == 1 and x_dim.dims != y_dim.dims:
```

Comment:
It seems this will not create the meshgrid in the (rare) special case where x_dims.dims == y_dims.dims, hence the domain is of square shape. Could testing for equal spacing of values in x_dim, y_dim do the trick?

Reply:
I'm not sure I understand. If `x_dim` and `y_dim` have a unique common dimension, then this is _not_ a meshgrid case, no? In that case, it's a series of points, not a grid. The meshgrid is useful when we have 1D specification of the two independent coordinates, which implies `x_dim.dims != y_dim.dims`.

Reply:
My thinking was that there could be a domain that has a square shape AND vectors for lat/lon requiring to generate the meshgrid. I can't think of data having a domain like that, so my comment is of neglectable practical relevance. It could exist, hypothetically.

Reply:
Oh! you mean something like:
```python
lat = [0, 1, 2, 0, 1, 2, 0, 1, 2]
lon = [0, 0, 0, 1, 1, 1, 2, 2, 2]
```
?

This could exist, but `np.meshgrid` wouldn't understand it correctly I think.

Also, the goal here is to keep the data in the same shape as the input. Re-squaring the previous example would defeat that purpose : a vector would go in and a matrix would come out.